test/e2e/framework/helpers/yunikorn/rest_api_utils.go

```diff
@@ -420,21 +420,21 @@ func GetFailedHealthChecks() (string, error) {
 }
 
 func (c *RClient) GetQueue(partition string, queueName string) (*dao.PartitionQueueDAOInfo, error) {
-	queues, err := c.GetQueues(partition)
+	req, err := c.newRequest("GET", fmt.Sprintf(configmanager.QueuePath, partition, queueName), nil)
 	if err != nil {
 		return nil, err
 	}
-	if queueName == "root" {
-		return queues, nil
-	}
+	// Include the query string 'subtree' to retrieve all queue children by default for better convenience in E2E tests
```

Comment:
My main intention was to make it easier for anyone who might use this utility method in the future. If a test case needs to check the queue's children and finds that this method doesn't retrieve them, it could be confusing, I think.

If this concern is unnecessary, I'm happy to remove this code. Or maybe `GetQueue()` could have an additional parameter: `bool withChildren` to remind callers and avoid potential confusion.

Reply:
> could have an additional parameter: bool withChildren to remind callers and avoid potential confusion.

that is good to me